backend/src/sms/services/sms-callback.service.ts

```diff
@@ -94,7 +114,81 @@ const parseEvent = async (req: Request): Promise<void> => {
     )
   }
 }
+
+// for future context: we only care about when the twilio callback is a finalized status
+// we don't want to update the status pre-emptively until we receive the finalized status
+// initial state --> UNSENT
+const parseTransactionalEvent = async (req: Request): Promise<void> => {
+  const {
+    MessageStatus: twilioMessageStatus,
```

Comment:
**[nit/change request]** know this part is consistent with existing code, but is it possible to use types from Twilio's library to make this part type-safe? 

or if that is too much of a hassle, maybe add a comment here to highlight this part is not type-safe and any changes here should be treated with caution

Reply:
added the comment.

P.S. there is much to be desired for a cleaner twilio integration but yes, there is only that much refactoring i can do in one PR before it becomes a risk, and better to have a separate PR to properly clean up twilio integration. The first step of moving twilio to shared folder is already gonna be huge in this refactoring should we decide to do it.

Reply:
agreed!